File: app/src/main/java/com/yao/zhihudaily/ui/theme/ThemeStoryAdapter.java
package com.yao.zhihudaily.ui.theme;

import android.app.Activity;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.yao.zhihudaily.R;
import com.yao.zhihudaily.model.ThemeStory;
import com.yao.zhihudaily.tool.OnItemClickListener;
import com.yao.zhihudaily.ui.NewsDetailActivity;

import java.util.ArrayList;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;

/**
 *
 * @author Administrator
 * @date 2016/9/10
 */
public class ThemeStoryAdapter extends RecyclerView.Adapter<ThemeStoryAdapter.StoryHolder> {

    private static final int STORY = 0;
    private static final int STORY_WITH_IMAGE = 1;

    private Activity aty;
    private ArrayList<ThemeStory> stories = new ArrayList<>();
    private OnItemClickListener listener = new OnItemClickListener() {
        @Override
        public void onItemClick(int pos) {
            ThemeStory themeStory = stories.get(pos);
            Intent intent = new Intent(aty, NewsDetailActivity.class);
            intent.putExtra("id", themeStory.getId());
            intent.putExtra("id", themeStory.getId());
            aty.startActivity(intent);
        }
    };

    public ThemeStoryAdapter(Activity aty) {
        this.aty = aty;
    }

    public void addList(ArrayList<ThemeStory> stories) {
        this.stories.addAll(stories);
    }

    @NonNull
    @Override
    public StoryHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        if (viewType == STORY) {
            return new StoryHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_theme_story, parent, false), STORY);
        } else {
            return new StoryHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_theme_story_with_image, parent, false), STORY_WITH_IMAGE);
        }
    }

    @Override
    public void onBindViewHolder(@NonNull StoryHolder holder, int position) {
        ThemeStory themeStory = stories.get(position);
        holder.tvTitle.setText(themeStory.getTitle());
        if (themeStory.getImages() != null && holder.iv != null) {
            Glide.with(aty).load(themeStory.getImages().get(0)).placeholder(R.mipmap.ic_launcher).into(holder.iv);
        }
        holder.itemView.setTag(position);
        holder.itemView.setOnClickListener(listener);

    }

    @Override
    public int getItemViewType(int position) {
        ThemeStory themeStory = stories.get(position);
        if (themeStory.getImages() == null) {
            return STORY;
        } else {
            return STORY_WITH_IMAGE;
        }
    }

    @Override
    public int getItemCount() {
        return stories.size();
    }

    class StoryHolder extends RecyclerView.ViewHolder {

        @Nullable
        @BindView(R.id.iv_splash)
        ImageView iv;
        @BindView(R.id.tv_title)
        TextView tvTitle;

        public StoryHolder(View view, int type) {
            super(view);
            ButterKnife.bind(this, itemView);
        }
    }


}

File: app/src/main/java/com/yao/zhihudaily/ui/theme/ThemeStoryAdapter.kt
package com.yao.zhihudaily.ui.theme

import android.app.Activity
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.yao.zhihudaily.R
import com.yao.zhihudaily.model.ThemeStory
import com.yao.zhihudaily.tool.Constant
import com.yao.zhihudaily.tool.OnItemClickListener
import com.yao.zhihudaily.ui.NewsDetailActivity
import kotlinx.android.synthetic.main.item_theme_story_with_image.view.*
import java.util.*

/**
 *
 * @author Yao
 * @date 2016/9/10
 */
class ThemeStoryAdapter(private val aty: Activity) : RecyclerView.Adapter<ThemeStoryAdapter.StoryHolder>() {
    private val stories = ArrayList<ThemeStory>()
    private val mOnItemClickListener = object : OnItemClickListener() {
        override fun onItemClick(pos: Int) {
            val themeStory = stories[pos]
            val intent = Intent(aty, NewsDetailActivity::class.java)
            intent.putExtra(Constant.ID, themeStory.id)
            intent.putExtra(Constant.ID, themeStory.id)
            aty.startActivity(intent)
        }
    }

    fun addList(stories: ArrayList<ThemeStory>) {
        this.stories.addAll(stories)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StoryHolder {
        lateinit var itemView: View
        lateinit var holder: RecyclerView.ViewHolder
        if (viewType == STORY) {
            itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_theme_story, parent, false)
            holder = StoryHolder(itemView, STORY)
        } else {
            itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_theme_story_with_image, parent, false)
            holder = StoryHolder(itemView, STORY_WITH_IMAGE)
        }
        itemView.setOnClickListener {
            mOnItemClickListener.onItemClick(holder.layoutPosition)
        }
        return holder
    }

    override fun onBindViewHolder(holder: StoryHolder, position: Int) {
        val themeStory = stories[position]
        holder.tvTitle.text = themeStory.title
        if (themeStory.images != null) {
            Glide.with(aty).load(themeStory.images!![0]).placeholder(R.mipmap.ic_launcher).into(holder.ivPic)
        }
    }

    override fun getItemViewType(position: Int): Int {
        val themeStory = stories[position]
        return if (themeStory.images == null) {
            STORY
        } else {
            STORY_WITH_IMAGE
        }
    }

    override fun getItemCount(): Int {
        return stories.size
    }

    inner class StoryHolder(view: View, type: Int) : RecyclerView.ViewHolder(view) {

        val ivPic: ImageView = view.iv_pic
        val tvTitle: TextView = view.tv_title
    }

    companion object {

        private val STORY = 0
        private val STORY_WITH_IMAGE = 1
    }


}
